### DISCUSSION TITLE
Different File Extensions

### DISCUSSION THREAD

**Discussion by sthendev** [scalameta/metals#3681] [Q&A] [answered]
Hi,

My project has scala files that don't conform to the normal *.scala, *.sbt, *.sc file extensions. Is there any way that I can tell the extension that files with my custom file extension are still scala files? Or perhaps the extension can pick up from vscode that I have associated this file extension with "Scala".

Any help is appreciated.

Thanks.

**Comment by sthendev** [comment, marked as answer]
Apologies, associating the file with Scala does appear to be working. I was having a different connection issue instead.

**Reply by tgodzik** [reply]
I think files with custom extensions might not work well in Metals or other build tools. It's probably best to use one of the existing extensions.
